google/cloud/spanner/value.h

```diff
@@ -137,7 +200,11 @@ class Value {
    */
   template <typename T>
   bool is() const {
-    return ProtoEqual(type_, MakeTypeProto(T{}));
+    google::protobuf::util::MessageDifferencer diff;
```

Comment:
I'm open to a change like this, but a couple days ago I just changed this the opposite direction: From an out-of-line function into the body of is. My reasoning was that this is slightly weird proto equality logic that only applies to this specific `is<T>()` function. An out-of-line function should likely not be called from anywhere else.

The reason this is weird is that this comparison needs to ignore the struct's "name" field. That's not always an OK thing to do --  in most cases, it would be an error to ignore the name field. However, specifically in the case of `is<T>()` we *must* ignore name, because `is<T>()` only gets the C++ type (`T`), it does not have access to the "name" fields.

Therefore, I thought it made sense to put that logic specifically in the body of `is<T>()`. However, it would be fine to factor that logic out into a very clearly named function like `EqualTypeProtoIgnoringName()`? Thoughts? If you're in favor of this, I'll happily make the change.

Reply:
Nevermind. On second thought, I think you're right. Fixing this in https://github.com/googleapis/google-cloud-cpp-spanner/pull/94